core/src/main/java/me/leoko/advancedban/manager/PunishmentManager.java

```diff
@@ -126,12 +127,33 @@ public void discard(String name) {
      * @return the punishments
      */
     public List<Punishment> getPunishments(String target, PunishmentType put, boolean current) {
+        return put == null ? getPunishments(target, current, PunishmentType.values()) : getPunishments(target, current, put);
+    }
+
+    /**
+     * Get all punishments which belong to the given uuid or ip.
+     *
+     * @param target  the uuid or ip to search for
+     * @param current if only active punishments should be included.
+     * @param put     an array of the basic punishment type to search for ({@link PunishmentType#BAN} would also include Tempbans).
+     *      *                Use <code>null</code> to search for all punishments.
+     * @return the punishments
+     */
+    public List<Punishment> getPunishments(String target, boolean current, PunishmentType ... put ) {
```

Comment:
While I really like the spread operator I would suggest keeping the same order of arguments and using an array instead. This provides more consistency.

So `getPunishments(String target, PunishmentType[] punishmentTypes, boolean current)`

Reply:
I actually wanted to but that provided some issues with the ambiguousness since i still would be passing a null from the history I cant @notnull it